### PR TITLE
Make the artifact docs more clear

### PR DESCRIPTION
**Pull request by plaindocs**
To address https://github.com/buildkite/docs/issues/705

- [x] link to H3 examples (waiting on implementation of #708 )
- [x] downloads
- [x] double check `/docs/agent/v3/cli-artifact` 

The `v3/cli` docs seem ok as is, and this is an improvement on what we have in the pipeline doc.

### CODE REVIEW

File: pages/pipelines/artifacts.md.erb [hunk 1]
```diff
@@ -6,7 +6,25 @@ In this guide we’ll walk through using the Buildkite Agent’s [artifact suppo
 
 ## Uploading Artifacts
 
-To automatically upload artifacts created by a build step, set the `artifact_paths` attribute on your command step. In this example, any artifacts that are created in the `logs` or `coverage` directories, as well as any of their subdirectories, will be uploaded automatically:
+How you define your build pipeline determines how you automatically upload artifacts created by build steps:
```
Comment (harrietgrace) [inline pages/pipelines/artifacts.md.erb:9]:
```suggestion
How you define your build pipeline determines how you configure the automatic upload for artifacts created by build steps:
```

File: pages/pipelines/artifacts.md.erb [hunk 2]
```diff
@@ -6,7 +6,25 @@ In this guide we’ll walk through using the Buildkite Agent’s [artifact suppo
...
+
+* pipelines defined in the Buildkite UI use the ["Automatic Artifact Uploading"](#uploading-artifacts-automatic-artifact-uploading) option.
```
Comment (harrietgrace) [inline pages/pipelines/artifacts.md.erb:11]: Could we pull out 'Pipelines defined in' to the top of the list, then have each dot point start from there? Also, we tend to not do full stops at the end of dot points unless there're multiple sentences 😊

Comment (plaindocs) [reply pages/pipelines/artifacts.md.erb:11]: Thanks for patiently walking me through the style guide. 

I tried pulling out the "Defined in", but i don't really like it. Both sentences end up super fragmented. Let's chat about the broader layout tomorrow.

**Comment by plaindocs** [comment]
@harrietgrace I think this addresses what we chatted about.

**Pull request merged**